### DISCUSSION TITLE
Ability for annotations to also be links.

### DISCUSSION THREAD

**Discussion by UltraInstinct05** [squidfunk/mkdocs-material#3989] [Ideas]
## Feature description

Currently, [annotations](https://squidfunk.github.io/mkdocs-material/reference/annotations/#using-annotations) are used for showing tooltips. I would like to request a feature similar to this, but instead of showing a tooltip, I wanted to make them links instead i.e. clicking on an annotation will navigate to target link.

## What's your use case for the feature

Taking the base case from here https://github.com/squidfunk/mkdocs-material/issues/3976#issuecomment-1146546243
Currently, you cannot have markdown links within code blocks, but annotations provide a neat way to inject little icons in the code block. If that is coupled with a feature that makes annotations, "links", I could simply use annotations as links to given tabs using the latest https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#anchor-links feature to fulfill my case.

**Comment by squidfunk** [comment]
We shouldn't be adding more semantics to annotations – annotations should always open a tooltip. You can add a link to this tooltip. Otherwise, it's pretty unclear for the user what will happen when clicking an annotation, which is considered bad UX. Same UI components = same behavior, as a general rule of thumb.

**Reply by UltraInstinct05** [reply]
I guess if not allowing annotations to also act as links, perhaps a separate component (which shows, let's say a link icon instead of a plus icon, indicating it won't display a tool tip, but rather it navigates to a link) ? Right now adding links in tool tips is still a 2 click process (Click the annotation and then click the link in the tool tip). Anyways, I can only think of 2 use cases for clickable annotations

- Showing a tool tip.
- Making it a clickable link.

A separate icon can, I feel, clearly distinguish between displaying tool tips and links IMHO, if the concern is "what will happen when clicking an annotation"

**Reply by squidfunk** [reply]
What we could add, in theory, is a URL parser that detects URLs inside comments and automatically converts them to hyperlinks, but this can be easily done via theme extension. You can follow the source of how annotations are implemented and do so via custom JavaScript. If it proves to be working well, we may include it into `master` if other users find that to be useful:

https://github.com/squidfunk/mkdocs-material/blob/30dcb66c710b960d048ae6de5cfdd97782a714fd/src/assets/javascripts/components/content/annotation/list/index.ts#L119-L127
